internal/support/scheduler/router.go

```diff
@@ -0,0 +1,231 @@
+//
+// Copyright (c) 2018 Tencent
+//
+// Copyright (c) 2017 Dell Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package scheduler
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/edgexfoundry/edgex-go/pkg/models"
+	mux "github.com/gorilla/mux"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+func LoadRestRoutes() http.Handler {
+	r := mux.NewRouter()
+	b := r.PathPrefix("/api/v1").Subrouter()
+
+	//ping
+	b.HandleFunc("/ping", ping).Methods(http.MethodGet)
+
+	// meta
+	b.HandleFunc("/info/{name}", info).Methods(http.MethodGet)
+
+	// callback
+	b.HandleFunc("/callbacks", addCallbackAlert).Methods(http.MethodPost)
+	b.HandleFunc("/callbacks", updateCallbackAlert).Methods(http.MethodPut)
+	b.HandleFunc("/callbacks", removeCallbackAlert).Methods(http.MethodDelete)
+
+	return r
+}
+
+const (
+	ContentTypeKey       = "Content-Type"
+	ContentTypeJsonValue = "application/json; charset=utf-8"
+	ContentLengthKey     = "Content-Length"
+)
+
+func ping(rw http.ResponseWriter, req *http.Request) {
+	rw.Header().Set(ContentTypeKey, ContentTypeJsonValue)
+	rw.WriteHeader(http.StatusOK)
+	str := `{"value" : "pong"}`
+	io.WriteString(rw, str)
+}
+
+func info(rw http.ResponseWriter, req *http.Request) {
+	rw.Header().Set(ContentTypeKey, ContentTypeJsonValue)
+	rw.WriteHeader(http.StatusOK)
+
+	vars := mux.Vars(req)
+
+	scheduleEvent, err := schedulerClient.QueryScheduleWithName(vars["name"])
+	if err != nil {
```

Comment:
After fixing the bug which I called out on the main.go file referring to the wrong routes being loaded, I then used Postman to test this method. I issued the following GET `http://localhost:48085/api/v1/info/test` and received a panic. The panic has to do with the schedulerClient not being initialized anywhere in the service and so every function that uses the schedulerClient will panic. Further, what is the intent here? The scheduler service should not be using its client to call itself. Should it be using `pkg/clients/metadata/schedule.go` instead?

```
INFO: 2018/10/06 18:33:39 This is the Support Scheduler Microservice
INFO: 2018/10/06 18:33:39 Service started in: 15.566394ms
INFO: 2018/10/06 18:33:39 Listening on port: 48085
2018/10/06 18:33:45 http: panic serving [::1]:56923: runtime error: invalid memory address or nil pointer dereference
goroutine 6 [running]:
net/http.(*conn).serve.func1(0xc0001cc0a0)
	/usr/local/go/src/net/http/server.go:1746 +0xd0
panic(0x1339080, 0x1636c70)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/edgexfoundry/edgex-go/internal/support/scheduler.info(0x1402c00, 0xc0001d6000, 0xc000168900)
	/Users/tconn/go/src/github.com/edgexfoundry/edgex-go/internal/support/scheduler/router.go:57 +0xe2
net/http.HandlerFunc.ServeHTTP(0x13b8768, 0x1402c00, 0xc0001d6000, 0xc000168900)
	/usr/local/go/src/net/http/server.go:1964 +0x44
github.com/edgexfoundry/edgex-go/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc0000ba1e0, 0x1402c00, 0xc0001d6000, 0xc000168900)
	/Users/tconn/go/src/github.com/edgexfoundry/edgex-go/vendor/github.com/gorilla/mux/mux.go:114 +0xe0
github.com/edgexfoundry/edgex-go/vendor/github.com/gorilla/context.ClearHandler.func1(0x1402c00, 0xc0001d6000, 0xc0001d0100)
	/Users/tconn/go/src/github.com/edgexfoundry/edgex-go/vendor/github.com/gorilla/context/context.go:141 +0x8b
net/http.HandlerFunc.ServeHTTP(0xc0000ae920, 0x1402c00, 0xc0001d6000, 0xc0001d0100)
	/usr/local/go/src/net/http/server.go:1964 +0x44
net/http.serverHandler.ServeHTTP(0xc000099450, 0x1402c00, 0xc0001d6000, 0xc0001d0100)
	/usr/local/go/src/net/http/server.go:2741 +0xab
net/http.(*conn).serve(0xc0001cc0a0, 0x1403000, 0xc000020240)
	/usr/local/go/src/net/http/server.go:1847 +0x646
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2851 +0x2f5
```

Reply:
Fixed this by removing the client code and adding references to the scheduler functions.
Updated the mux to use the bone style parameters.  
Updated the routes as well to use the style similar to the logging service.